Content.Server/Supermatter/Systems/SupermatterSystem.cs

```diff
@@ -0,0 +1,546 @@
+            }
+            else
+            {
+                //Absorbed gas from surrounding area
+                var absorbedGas = mixture.Remove(sMcomponent.GasEfficiency * mixture.TotalMoles);
+                var absorbedTotalMoles = absorbedGas.TotalMoles;
+
+                //Mols start to have a positive effect on damage after 350
+                sMcomponent.Damage = (float) Math.Max(
+                    sMcomponent.Damage +
+                    Math.Max(
+                        Math.Clamp(absorbedTotalMoles / 200, 0.5, 1) * absorbedGas.Temperature -
+                        (Atmospherics.T0C + sMcomponent.HeatPenaltyThreshold) * sMcomponent.DynamicHeatResistance,
+                        0) * sMcomponent.MoleHeatPenalty / 150 * sMcomponent.DamageIncreaseMultiplier, 0);
+
+                //Power only starts affecting damage when it is above 5000
+                sMcomponent.Damage =
+                    Math.Max(
+                        sMcomponent.Damage +
+                        Math.Max(sMcomponent.Power - sMcomponent.PowerPenaltyThreshold, 0) / 500 *
+                        sMcomponent.DamageIncreaseMultiplier, 0);
+
+                //Molar count only starts affecting damage when it is above 1800
+                sMcomponent.Damage =
+                    Math.Max(
+                        sMcomponent.Damage + Math.Max(absorbedTotalMoles - sMcomponent.MolePenaltyThreshold, 0) / 80 *
+                        sMcomponent.DamageIncreaseMultiplier, 0);
+
+                //There might be a way to integrate healing and hurting via heat
+                //healing damage
+                if (absorbedTotalMoles < sMcomponent.MolePenaltyThreshold)
+                {
+                    //Only has a net positive effect when the temp is below 313.15, heals up to 2 damage. Psycologists increase this temp min by up to 45
+                    sMcomponent.Damage =
+                        Math.Max(
+                            sMcomponent.Damage +
+                            Math.Min(absorbedGas.Temperature - (Atmospherics.T0C + sMcomponent.HeatPenaltyThreshold),
+                                0) / 150, 0);
+                }
+
+                //if there are space tiles next to SM
+                //TODO: change moles out for checking if adjacent tiles exist
+                foreach (var ind in _atmosphere.GetAdjacentTileMixtures(xform.GridUid.Value, indices))
+                {
+                    if (ind.TotalMoles != 0)
+                        continue;
+
+                    var integrity = GetIntegrity(sMcomponent.Damage, sMcomponent.ExplosionPoint);
+
+                    var factor = integrity switch
+                    {
+                        < 10 => 0.0005f,
+                        < 25 => 0.0009f,
+                        < 45 => 0.005f,
+                        < 75 => 0.002f,
+                        _    => 0f
+                    };
```

Comment:
Looks like there's a fair few hardcoded values in here, but I'm not parsing all the code this second.